### PR TITLE
Add sleep test plan

### PR DESCRIPTION
**Pull request by tomeuv**
Add a start for a test plan that checks that system power states work,
along with RTC wake ups.

### CODE REVIEW

**Comment by mattface** [comment]
Your template files need to be named with "sleep" in them, see NFS or "simple" for examples. This is so that they can be identified in the device_map.

**Comment by gctucker** [comment]
Also I suppose it would be good to have an extra patch to add this test plan to at least one board (say, rk3288_veyron_jaq) in `device_map.py`.  That way we could test it on staging, then add it to more boards gradually.

**Comment by khilman** [comment]
You're adding it to the generic ramdisk, do we have rtcwake in the default ramdisk?

**Comment by tomeuv** [comment]
> You're adding it to the generic ramdisk, do we have rtcwake in the default ramdisk?

Yep. You can see the logs at https://lava.collabora.co.uk/scheduler/job/1078742

File: templates/sleep/TODO [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- skip power states not in /sys/power/state
+- probably loop a few times (at least two)
+- may need to configure RTCs other than rtc0
+- find a way to not replicate the actual test cases in each template
```
Comment (gctucker) [inline templates/sleep/TODO]: You can (and probably should) already fix that with a separate template fragment, say `templates/sleep/sleep.jinja2` which will contain the test definition.  Then you can include that in each template variant with the jinja2 `include` statement.

**Review by gctucker — COMMENTED** [review]
We'll soon need to consolidate how test plans are managed, with templates and entries in the device map.  For this one, we can still stick to the current state of things but avoiding duplication of the test definition in the templates would already be a good first step.

File: templates/sleep/TODO (hunk 1 above)
Comment (tomeuv) [reply templates/sleep/TODO]: Done, thanks.

**Review by khilman — APPROVED** [review]

File: templates/sleep/generic-depthcharge-tftp-ramdisk-sleep-template.jinja2 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{% extends 'base/kernel-ci-base-tftp-deploy.jinja2' %}
```
Comment (gctucker) [inline templates/sleep/generic-depthcharge-tftp-ramdisk-sleep-template.jinja2]: Maybe extend `generic-depthcharge-tftp-ramdisk-template.jinja2` instead and remove the boot commands below?

Comment (tomeuv) [reply templates/sleep/generic-depthcharge-tftp-ramdisk-sleep-template.jinja2]: Yep, that worked fine, thanks!

**Pull request merged**